### DISCUSSION TITLE
BUG: Max concurrent downloads, makes additional downloads get stuck and cannot be stopped/removed

### DISCUSSION THREAD

**Discussion by FrisoDB** [aclap-dev/video-downloadhelper#1060] [Help and Q&A]
### Your question

The concurrent download limit (Max concurrent downloads) is set at 6 by default. If a 7th download is added, this one does not start without warning. If the limit is raised to 10 or 15, the download still does not start but remains stuck, with the faint red spinning circle next to it in the downloads overview in the pop-up window. There is no button to stop this download, only restarting the whole browser seems to work. Also adding new downloads at this point does not work.

It would be great if:
- there is a warning when reaching the concurrent download limit
- stuck downloads can be stopped
- new downloads can be added directly after raising the limit

### Addon & CoApp version and details

version: 9.0.2.12
target: mozilla
channel: stable
lang: nl
coapp: {"found":true,"path":"C:\\Program Files\\DownloadHelper CoApp\\vdhcoapp.exe","version":"2.0.19","new_version":false}
license: {"unset":true}
platform: x86-64 win
UA: Mozilla/5.0 (Windows NT 10.0; Win64; x64; rv:128.0) Gecko/20100101 Firefox/128.0
{
    "used_sidebar_button": true,
    "used_popup_button": true,
    "concurrent_downloads_max": 6,
    "view_options": {
        "all_tabs": false,
        "low_quality": false,
        "sort_by_status": true,
        "sort_reverse": false,
        "show_button_clean": true,
        "show_button_clean_all": true,
        "show_button_convert_local": false,
        "hide_downloaded": false
    },
    "use_wide_ui": false,
    "use_legacy_ui": false,
    "use_sidebar": false,
    "open_count_store": 17,
    "successfull_dl": 50
}

**Comment by GrampaWildWilly** [comment]
I do not work for VDH.  I'm just another user.

I'm running Windows 11 Pro 64-bit version 23H2 build 22631.3958, Firefox 128.0.3 64-bit, licensed VDH 9.0.2.14a1 beta, CoApp 2.0.19.  Note that.  Beta.  You might want to try that.  Maybe it will fix your problem, maybe not.  But it can't hurt to try it.  You (everybody) should be running the beta.  Unlike other software, whose beta versions are bug-filled experimental versions, the VDH beta is just an interim release that includes bug fixes.  People should really install the beta.  There's no sense in encountering a problem that's already been solved & being told to install the beta because the fix is in there.  So you should always run the beta version of VDH.  I haven't run a general availability version of VDH in at least 4 years.  I can think of only 1 time in all those years when a beta was so bad I couldn't use it.  That was not recently so don't ask me which one it was.  I'm pretty sure it was a VDH 7 beta but I couldn't tell you the exact release number.  That problem got fixed within a day anyway.  In any case, try the beta.

I have never encountered the problem you're describing.  Mind you, I'm running with VDH set to permit 32 concurrent downloads, a limit I've never hit.  In a situation where you have a zombie download like you got, it's not necessary to close all browser windows.  You can go into the `More settings` section of VDH settings, scroll to the bottom, & click the `Reload addon` button.  It will clear everything, stuck downloads, completed downloads, downloads in progress.  Downloads in progress.  Anything you might be successfully downloading will be abruptly terminated.  The result files will be useless.  You will have to do those downloads over again from the beginning.  So I would wait for inflight downloads to complete first, then click the `Reload addon` button.

**Reply by FrisoDB** [reply]
Good that you have not had this issue. Basically you are describing the exact reason why a reset/reload of the add-on is unwanted. Furthermore, if you ever get a broken file from VDH, it may be truncated such that the moov part is missing. untrunc_x64 (anthwlock) may be useful for you, or digital video repair (risingresearch).

Anyways, let's see if we can get someone from development to notice this bug, so that it can be solved. It has been a longer existing issue, which reappears after updating sometimes. Very unpleasant to deal with and for anyone less inclined to investigate probably a final sign of instability. Would you know someone to notify?

**Reply by GrampaWildWilly** [reply]
Posting here is the way to communicate with the developer.  He is @paulrouget Paul.  He participates in this forum.  This is the best you can do.

**Comment by sekullbe** [comment]
I wouldn't call those 'stuck' downloads, just 'queued', but I would like the ability to cancel or re-order them. The old UI allowed this.
